mlir/include/mlir/IR/CommonTypeConstraints.td

```diff
@@ -908,6 +908,11 @@ def BoolLike : TypeOrContainer<I1, "bool-like">;
 
 def BoolLikeOfAnyRank : TypeOrContainerOfAnyRank<I1, "bool-like">;
 
+// Type constraint for signless-integer-like types: signless integers, 
+// vectors of signless integers or tensors of signless integers.
+def SignlessInteger : TypeOrValueSemanticsContainer<
```

Comment:
This is a bit confusing because there is also `SignlessIntegerLike` below.

I would change it as follows:
- Rename `SignlessIntegerLike` to `SignlessIntegerOrIndexLike`.
- Rename `SignlessIntegerLikeOfAnyRank` to `SignlessIntegerOrIndexLikeOfAnyRank`
- Call this new definition `SignlessIntegerLike` instead of `SignlessInteger`.

Reply:
Thanks for comment, I rename these constraints.